### PR TITLE
[#174824788] subscription feed on user delete

### PR DESCRIPTION
**Pull request by balanza**
Add a record of type `UNSUBSCRIBE` to the subscription feed when a user is deleted in his/her claim to erasure.

A connection to the SubscriptionsFeed storage table is added to the current service.

### Env variables
The following environment variables have been added:

| Variable name                    | Description                                                                                      | type   |
|----------------------------------|--------------------------------------------------------------------------------------------------|--------|
| QueueStorageConnection           | Storage connection string for subscription feed                                                  | string |
| SUBSCRIPTIONS_FEED_TABLE         | Table name for the Subscriptions Feed in the storage                                             | string |

### CODE REVIEW

File: UpdateSubscriptionsFeedActivity/index.ts [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+      subscriptionKind: t.literal("SERVICE")
+    })
+  ])
+]);
+
+export type Input = t.TypeOf<typeof Input>;
+
+// When the function starts, attempt to create the table if it does not exist
+// Note that we cannot log anything just yet since we don't have a Context
+tableService.createTableIfNotExists(subscriptionsFeedTable, () => 0);
```
Comment (gunzip) [inline UpdateSubscriptionsFeedActivity/index.ts:61]: remove these lines (they're useless even in io-fn-app)

File: env.example [hunk 2]
```diff
@@ -53,3 +53,6 @@ NODE_TLS_REJECT_UNAUTHORIZED=0
 languageWorkers__node__arguments="--inspect=0.0.0.0:5861"
 
 # AzureWebJobs.UpdateVisibleServicesCache.Disabled=true
+
+QueueStorageConnection=<STORAGE_CONNECTION_STRING>
```
Comment (gunzip) [inline env.example:57]: can we rename `QueueStorageConnection` to `SubscriptionFeedStorageConnection` ?

**Review by gunzip — CHANGES_REQUESTED** [review]

File: env.example (hunk 2 above)
Comment (balanza) [reply env.example:57]: Of course, just note that it's the same value used here https://github.com/pagopa/io-functions-app/blob/master/UpdateSubscriptionsFeedActivity/index.ts#L18

**Review by gunzip — APPROVED** [review]
lgtm @francescopersico can we merge this?

File: UpdateSubscriptionsFeedActivity/index.ts [hunk 3]
```diff
@@ -0,0 +1,176 @@
+﻿import * as crypto from "crypto";
```
Comment (francescopersico) [inline UpdateSubscriptionsFeedActivity/index.ts:1]: Do we really want to maintain all this logic when we just want to add an UNSUBSCRIBE event in the table?

Comment (gunzip) [reply UpdateSubscriptionsFeedActivity/index.ts:1]: May you mark the parts that you think are redundant? I've asked the same question myself, then I noticed that the intent here was to have a general purpose activity (as the name is indeed 'UpdateSubscriptionsFeedActivity' and not 'UnsubscribeUser'). If we switch this to `UnsubscribeUser` we still need to keep 80% of the logic and you only gain about 10 lines less getting rid of the 'subscribe' branch.

Comment (francescopersico) [reply UpdateSubscriptionsFeedActivity/index.ts:1]: The risk of duplicating the "subscribe" code here is that you must maintain it in sync if changes happens.
About sharing the code inside io-functions-commons i am more convinced that subscriptions must be managed by a specific/separate service.

Comment (gunzip) [reply UpdateSubscriptionsFeedActivity/index.ts:1]: what do you suggest in the meantime?

Comment (francescopersico) [reply UpdateSubscriptionsFeedActivity/index.ts:1]: It is ok for me to merge as is.

**Review by francescopersico — APPROVED** [review]

**Pull request merged**